pkgs/by-name/ta/tabby-terminal/package.nix

```diff
@@ -0,0 +1,215 @@
+      repo = "tabby";
+      rev = "v${version}";
+      hash = "sha256-l3Hyt3e1edJjb7vL66JDllkpHwTU5efsNbgJ5Pcvn60=";
+      leaveDotGit = true;
+      deepClone = true;
+    };
+
+    meta = with lib; {
+      description = "A terminal for a more modern age";
+      homepage = "https://tabby.sh";
+      license = licenses.mit;
+      mainProgram = "tabby";
+      maintainers = with maintainers; [ geodic ];
+      platforms = [ "x86_64-linux" ];
+    };
+
+    # We need to fix the argv logic as tabby only handles it for the node executable, but we are running it with electron
+    patches = [ ./fix-argv-prefix-splice.patch ];
+
+    electronVersion = "29.4.6";
+
+    electronHeaders = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronVersion}/node-v${electronVersion}-headers.tar.gz";
+      hash = "sha256-e0qiEGI/1rWK72oLXfKtbbc8KmOGMhMx4IbT6MGAjms=";
+    };
+
+    electronHeadersSHA = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronVersion}/SHASUMS256.txt";
+      hash = "sha256-0mCnKIR3GS1TL+g0h+y2cAlqbBogf88WxN8xJvmFl4g=";
+    };
+
+    buildInputs = [
+      nodejs
+      python3
+      fontconfig
+      electron_29
+    ];
+
+    nativeBuildInputs = [
+      git
+      yarn
+      patch-package
+      prefetch-yarn-deps
+      fixup-yarn-lock
+      (python3.withPackages (python-pkgs: [
+        python-pkgs.distutils
+        python-pkgs.gyp
+      ]))
+      makeWrapper
+      http-server
+      pkg-config
+      libsecret
+      desktop-file-utils
+    ];
+
+    configurePhase =
+      ''
+        export buildDir=$PWD
+      ''
+      # Loop through all the yarn caches and install the deps for the respective package
+      + builtins.concatStringsSep "\n" (
+        map (cache: ''
+          cd ${cache.pkg}
+          export HOME=$PWD/yarn_home
+          fixup-yarn-lock yarn.lock
+          yarn config --offline set yarn-offline-mirror ${cache.cache}
+          echo "Installing deps for ${cache.pkg}"
+          yarn install --offline --prefer-offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+          patch-package
+          patchShebangs node_modules
+          echo "Done!"
+          cd $buildDir
+        '') yarnCaches
+      )
+      + ''
+        cd $buildDir/node_modules
+      ''
+      # Loop thought the "built in" plugins and link them to the node_modules
+      + builtins.concatStringsSep "\n" (
+        map (plugin: ''
+          ln -fs ../${plugin} ${plugin}
+        '') builtinPlugins
+      )
+      + ''
+        cd $buildDir
+      '';
+
+    buildPhase =
+      # Start a fake (and completely unnecessary) http-server to let electron-rebuild "download" the headers and hashes
+      ''
+        mkdir -p http-cache/v${electronVersion}
+        cp $electronHeaders http-cache/v${electronVersion}/node-v${electronVersion}-headers.tar.gz
+        cp $electronHeadersSHA http-cache/v${electronVersion}/SHASUMS256.txt
+        http-server http-cache &
```

Comment:
I don't think this is necessary. From a glance, `electron-rebuild` supports the `--dist-url` option which should make it avoid downloading Electron headers all the same.

Reply:
I tried to set the `--dist-url` to a file url, but it throws an error saying that it only supports HTTP(S)

Reply:
Ah, so it wants a *web* URL... Never mind then, maybe someone else would have a better idea